### PR TITLE
Use buttaface's android toolchain in docker

### PR DESCRIPTION
**Pull request by cobbal**

### CODE REVIEW

File: .github/workflows/integration-tests.yaml [hunk 1]
```diff
@@ -143,19 +160,83 @@ jobs:
...
+          ./gradlew publishToMavenLocal
+
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: maven-local
+          path: ~/.m2/repository
+
+  android-tests:
+    name: "android: integration tests"
```
Comment (mstokercricut) [inline .github/workflows/integration-tests.yaml:196]: Nit: Should look like the rest: 
`name: Android Integration Tests`

File: .github/workflows/integration-tests.yaml [hunk 2]
```diff
@@ -143,19 +160,83 @@ jobs:
           rm -rf $libdir
           mkdir -p $libdir
 
-          swift build --configuration release --product FishyJoesJavaRuntime
-          cp .build/release/libFishyJoesJavaRuntime.so $libdir
+          ./scripts/compile-kotlin-native-runtime.sh
+
+
+      - name: Build android FishyJoesJavaRuntime
+        uses: addnab/docker-run-action@4f65fabd2431ebc8d299f8e5a018d79a769ae185
```
Comment (mstokercricut) [inline .github/workflows/integration-tests.yaml:167]: Any concern about security on this one? Looks like it comes from a user, not GitHub...

**Review by mstokercricut — APPROVED** [review]
Love the android test app. Approved with nits.

File: .github/workflows/integration-tests.yaml (hunk 2 above)
Comment (cobbal) [reply .github/workflows/integration-tests.yaml:167]: I looked over the script, and that's why it's pinned to an exact commit

**Pull request merged**